crates/driver/src/infra/config/file/mod.rs

```diff
@@ -12,7 +12,8 @@ pub use load::load;
 
 #[serde_as]
 #[derive(Debug, Deserialize)]
-#[serde(rename_all = "kebab-case", deny_unknown_fields)]
+// TODO: add back `deny_unknown_fields` once we are ready to release the driver
```

Comment:
I'm honestly not so sure we want to add it back unless we can more precisely time the release with the config update. I'm sure we will keep adding parameters and while denying unknown fields is nice to keep the config minimal, maybe it's something that can run somewhat separately from the core deployment cycle.

Reply:
Also totally fine with dropping that entirely. 👌 
Theoretically we can also add a `--strict` CLI argument that controls that behavior. Not sure how useful that is, though.
Maybe we can just simply copy whatever config we currently have in the infra and run that with a local version that enables `deny_unknown_fields` to know which arguments we can throw out.

Reply:
> (for example, we have encountered issues where configuring SOME_PROPETRY instead of SOME_PROPERTY led us to silently run without a configuration option set for non trivial amounts of time, and in a way that was hard to debug)

That's a good point. The main goal of this PR is to make debugging syntax error easier (possible actually). I think your suggestion to use `TOML_TRACE_ERROR` works well enough that we can keep `deny_unknown_fields`.